### ISSUE TITLE
aria-roledescription overrides semantic roles of descendant elements

### ISSUE THREAD

**Issue by Wildebrew** [FreedomScientific/standards-support#288]
### Summary

When the `aria-roledescription` attribute is applied to a container element, Jaws will override the roles of all child elements.

Example:
Save the following code as an HTML file and run Jaws on it (with Chrome or Firefox):

```
<div role="region" aria-roledescription="carousel" aria-label="Our stuff">
<h3>Our awesome products</h3>
<p>We got a variety of stuff that you may or may not like, but you should buyit anyway because we say so.</p>

  <button>Next</button>
  <button>Previous</button>
</div>
```

### Expected result
* When you enter/exit the grouping Jaws should say "our stuff carousel".
* When you hit a heading inside the carousel Jaws should announce "Our awesome products, level 3 heading".
* If you navigate to the next/previous buttons (by tab or b or in browse mode) Jaws should say "previous button" "next button".

### Actual result

* Jaws does not announce the aria-roledescription value on the region as you enter/exit it.
* When you navigate to the heading Jaws will say "our awesome stuff, carousel".
* When you navigate to the buttons Jaws will say "previous carousel" / "next carousel".

### Example

One of the primary use cases for the `aria-roledescription` attribute is to mark landmark regions, e.g. slides.
See example 25 and 26 in the [aria-roledescription entry in the ARIA 1.1 spec](https://www.w3.org/TR/wai-aria-1.1/#aria-roledescription)

This approach is also recommended for carousel and slide containers in the [ARIA Authoring Practices carousel recommendation](http://www.w3.org/TR/wai-aria-practices-1.1/#carousel)

But when Jaws does not announce the aria-roledescription value where it is supposed to (on the grouping) and instead replaces the role of the descendnat elements with the value of the aria-roledescription, as it does today, this makes navigation for a Jaws user inside a region with an aria-roledescription difficult if not impossible.

All that Jaws should do with this attribute is to announce it on the element itself, not on its descendants.

I added a comment explaining same to [issue 154](https://github.com/FreedomScientific/VFO-standards-support/issues/154)
Issue 256 that also addresses aria-roledescription is a different scenario and I don't think it is directly related to this.

 
#### JAWS version and build number

Jaws 2019 (on Chrome 74 and Firefox 60-ESR, IE11 is a goner by now).

 
#### Operating System and version

Windows 7/Windows 10 (N/A really).

#### Browser and version:

Chrome 74
Firefox 60-ESR
IE11

**Comment by JAWS-test** [comment]
The problem only occurs when the aria-roledescription region is additionally labeled. If the aria label is removed, the output is correct with JAWS 2019 and IE 11, Firefox and Chrome.

**Comment by Wildebrew** [comment]
Aah, interesting.
It's still a bug in my opinion because the majority of the scenarios where you are doing this you want the container to have an accessible name (e.g. "slide 1", see the ARIA 1.1 examples).
But I had not thought of trying this without an accessible name, so great insight there.

**Comment by JAWS-test** [comment]
Of course it is a very serious bug, because the labeling is important and even mandatory for many elements.

**Comment by JAWS-test** [comment]
I checked the buttons with Inspect and couldn't see why the role wasn't output. In Inspect they are correctly displayed as buttons. So it doesn't seem to be a browser bug but a JAWS bug.

**Comment by stevefaulkner** [comment]
Closing this, please feel free to re-open if this still occurs with Latest (JAWS 2024).

**Comment by JAWS-test** [comment]
@stevefaulkner The problem continues to occur (current Chrome, JAWS 2024). But no longer with role=region, but still with role=toolbar, for example. In my opinion, the problem is very serious. However, there are still open tickets for this problem, so it is ok if it is closed here.
